npm/private/npm_link_package.bzl

```diff
@@ -336,26 +341,37 @@ def _impl_direct(ctx):
 
     package = ctx.attr.package if ctx.attr.package else ctx.attr.src[NpmLinkedPackageStoreInfo].package
 
-    output_files = []
-
     # symlink the package's path in the virtual store to the root of the node_modules
     # "node_modules/{package}" so it is available as a direct dependency
     root_symlink_path = paths.join("node_modules", package)
-    output_files = _make_symlink(ctx, root_symlink_path, virtual_store_directory)
 
-    result = [
+    files = _make_symlink(ctx, root_symlink_path, virtual_store_directory)
+    files_depset = depset(files)
+
+    transitive_files_depset = depset(files, transitive = [ctx.attr.src[NpmLinkedPackageStoreInfo].transitive_files])
+
+    providers = [
         DefaultInfo(
-            files = depset(output_files, transitive = [ctx.attr.src[DefaultInfo].files]),
+            files = transitive_files_depset,
+        ),
+        NpmLinkedPackageDirectInfo(
+            label = ctx.label,
+            link_package = ctx.label.package,
+            package = ctx.attr.src[NpmLinkedPackageStoreInfo].package,
+            version = ctx.attr.src[NpmLinkedPackageStoreInfo].version,
+            store_info = ctx.attr.src[NpmLinkedPackageStoreInfo],
+            files = files_depset,
+            transitive_files = transitive_files_depset,
         ),
         DeclarationInfo(
-            declarations = depset(output_files, transitive = [ctx.attr.src[DeclarationInfo].declarations]),
-            transitive_declarations = ctx.attr.src[DeclarationInfo].transitive_declarations,
+            declarations = depset(files, transitive = [ctx.attr.src[DeclarationInfo].declarations]),
+            transitive_declarations = depset(files, transitive = [ctx.attr.src[DeclarationInfo].transitive_declarations]),
```

Comment:
`files` is both a declaraction + transitive declaration?

Reply:
This is out of date now that the code has been refactored further